lib/liquid/standardfilters.rb

```diff
@@ -143,11 +143,19 @@ def sort_natural(input, property = nil)
       ary = InputIterator.new(input)
 
       if property.nil?
-        ary.sort { |a, b| a.casecmp(b) }
+        ary.sort { |a, b| a.to_s.casecmp(b.to_s) }
       elsif ary.empty? # The next two cases assume a non-empty array.
         []
       elsif ary.first.respond_to?(:[]) && !ary.first[property].nil?
-        ary.sort { |a, b| a[property].casecmp(b[property]) }
+        ary.sort do |a, b|
+          a = a[property]
+          b = b[property]
+          if a && b
+            a[property].to_s.casecmp(b[property].to_s)
```

Comment:
Hmm I'm curious why there's no failing test... this is equivalent to `a[property][property].to_s`

Reply:
There is no test coverage for that it seems.

Reply:
I added test coverage for this and fixed that issue